component-samples/demo/scripts/fdo_toolkit.sh

```diff
@@ -0,0 +1,282 @@
+    echo "y" | docker system prune -a --volumes
+    echo "Removed all containers, images and volumes"
+}
+
+isDockerComposeAtLeast() {
+    : ${1:?}
+    local minVersion=$1
+    if ! command -v docker-compose >/dev/null 2>&1; then
+        return 1   # it is not even installed
+    fi
+    # docker-compose is installed, check its version
+    lowerVersion=$(echo -e "$(docker-compose version --short)\n$minVersion" | sort -V | head -n1)
+    if [[ $lowerVersion == $minVersion ]]; then
+        return 0   # the installed version was >= minVersion
+    else
+        return 1
+    fi
+}
+
+install_dependencies() {
+
+echo "Installing all FDO dependencies"
+
+if java -version 2>&1 | grep version | grep -q 17.; then
+        echo "Found java 17"
+else
+        echo "Java 17 not found, installing it..."
+        apt-get update && apt-get install -y openjdk-17-jre-headless
+        chk $? 'installing java 17'
+fi
+
+
+if ! dpkg-query -s haveged > /dev/null 2>&1; then
+    echo "Haveged is required, installing it"
+    sudo apt-get install -y haveged
+    chk $? 'installing haveged'
+fi
+
+# If docker isn't installed, do that
+if ! command -v docker >/dev/null 2>&1; then
+    echo "Docker is required, installing it..."
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    chk $? 'adding docker repository key'
+    add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    chk $? 'adding docker repository'
+    apt-get install -y docker-ce docker-ce-cli containerd.io
+    chk $? 'installing docker'
+fi
+
+# If docker-compose isn't installed, or isn't at least 1.21.0 (when docker-compose.yml version 2.4 was introduced), then install/upgrade it
+# For the dependency on 1.21.0 or greater, see: https://docs.docker.com/compose/release-notes/
+minVersion=2.24.5
+if ! isDockerComposeAtLeast $minVersion; then
+    if [[ -f '/usr/bin/docker-compose' ]]; then
+        echo "Error: Need at least docker-compose $minVersion. A down-level version is currently installed, preventing us from installing the latest version. Uninstall docker-compose and rerun this script."
+        exit 2
+    fi
+#    echo "docker-compose is not installed or not at least version $minVersion, installing/upgrading it..."
+#    # Install docker-compose from its github repo, because that is the only way to get a recent enough version
+#    curl --progress-bar -L "https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose ## TODO link not working
+#    chk $? 'downloading docker-compose'
+#    chmod +x /usr/local/bin/docker-compose
+#    chk $? 'making docker-compose executable'
+#    ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+#    chk $? 'linking docker-compose to /usr/bin'
+fi
+
+
+echo "Upgraded and Installed all FDO dependencies"
+}
+
+
+update_scripts() {
+cd $1
```

Comment:
Better to check if such path or directory that is provided as argument exists and then change directory to the same